Update tile packs

TrevorMDev opened the pull request:
- Added custom configs storable per tile pack. This will allow users to customize each tile pack more.
- Added first per pack config, a visibility toggle. This allows users to hide the packs from the list.
![image](https://github.com/user-attachments/assets/da7b710d-a3bd-490d-8d71-414724b3fdb5)

- Added global filter that allows the user to control if they the list shows the visible, hidden, or all tile packs.
![image](https://github.com/user-attachments/assets/2ed1b7b1-1a01-47f5-be95-e9131a1be15b)
- Update README with useful information.

aHooder reviewed (approved):
All looks good here, but I have one minor suggestion. If you disable HTML for [your `TilePackPanel`'s `JLabel` here](https://github.com/TrevorMDev/tile-packs/blob/6940f003dd3092358a88fd4444d37e138677d614/src/main/java/com/tilepacks/ui/panel/TilePackPanel.java#L97) like so:
```java
packName = new JLabel();
packName.putClientProperty("html.disable", Boolean.TRUE);
packName.setText(tilePack.packName);
```
it won't be possible for your users to accidentally leak their IP by pasting malicious HTML into a custom pack name. Setting the text *after* the hint to disable HTML is important btw.

TrevorMDev commented:
Thanks for the suggestion, I've added it in. I also fixed a bug I found while testing it where newly added packs were enabled, but the point overlays were not being rendering.
